modules/system/primary-user.nix

```diff
@@ -19,6 +19,13 @@
       '';
     };
 
+    system.primaryUserHome = lib.mkOption {
+      internal = true;
+      type = lib.types.str;
+      default =
+        config.users.users.${config.system.primaryUser}.home or "/Users/${config.system.primaryUser}";
+    };
```

Comment:
Will this break anything if `config.system.primaryUser` is set to `null` (the default)? If I'm reading it correctly, it's only an issue if `primaryUserHome` is read by something, in which case `primaryUser` needs to be set anyway, but I wonder what the error message would look like. Maybe it's caught by `nix.nixPath` requiring `primaryUser` to be set? Not sure that works for `environment.darwinConfig` though

Reply:
It’ll be an evaluation error if anything depends on it. But, yes, that stuff should set `system.requiresPrimaryUser` (as `environment.darwinConfig` does) and therefore hopefully give a helpful error message before anything that uses it would get evaluated. `"${null}"` is also an error so I’m not sure allowing it to be `null` would improve anything. Maybe this could be a more descriptive `throw` in that case? But I haven’t triggered it so far in testing.

Reply:
Maybe
```nix
lib.warnIf (config.system.primaryUser == null) "`system.primaryUser` is unset but `system.primaryUserHome` is being read" config.users.users.${config.system.primaryUser}.home or "/Users/${config.system.primaryUser}";
```
?

On the other hand, it seems like this problem will only be hit due to a future mistake on our part (reading `primaryUserHome` in an option without adding that option to `requiresPrimaryUser`), so we could just leave this as is.

Reply:
Making it a `throwIf` might be a good idea, and seems pretty harmless. I’ll think about it. The main thing is that I wish we could make incorrect uses of `config.system.primaryUser` fail loudly in the same way, but that’s harder.